screenshot-testing-plugin/src/py/android_screenshot_tests/recorder.py

```diff
@@ -88,15 +90,22 @@ def _clean(self):
             shutil.rmtree(self._output)
         os.makedirs(self._output)
 
+    def _image_rms(self, image):
+        histogram = image.histogram()
+        squares = (value * ((idx % 256) ** 2) for idx, value in enumerate(histogram))
+        sum_of_squares = sum(squares)
+        rms = math.sqrt(sum_of_squares / float(image.size[0] * image.size[1]))
+        return rms
+
     def _is_image_same(self, file1, file2, failure_file):
         with Image.open(file1) as im1, Image.open(file2) as im2:
             diff_image = ImageChops.difference(im1.convert("RGB"), im2.convert("RGB"))
             try:
-                diff = diff_image.getbbox()
-                if diff is None and im1.size == im2.size:
+                if im1.size == im2.size and self._image_rms(diff_image) <= self._tolerance:
```

Comment:
Reversed conditions of checking size and difference, as checking size is O(1) and in case of failure, expression would not evaluate the other condition.